src/virtual_elements.js

```diff
@@ -275,13 +275,14 @@ var elementVoid = function(tag, key, statics, var_args) {
  * @param {...(function(string|number|boolean):string|number|boolean)} var_args
  *     Functions to format the value which are called only when the value has
  *     changed.
+ * @return {!Text} The corresponding Text Node.
  */
 var text = function(value, var_args) {
   if (process.env.NODE_ENV !== 'production') {
     assertNotInAttributes();
   }
 
-  var node = alignWithDOM('#text', null);
+  var node = /** @type {!Text}*/(alignWithDOM('#text', null));
```

Comment:
> Refactor alignWithDOM out to deal with Text and Element's differently which would then allow it to return the correct type from createTextNode directly

That one sounds the best.

> Leave them off and ignore the closure compiler (if thats the case I would question why bothering with a partial type annotations in the first place)

It's for the closure linter, which hadn't been giving us a problem yet. Try running `npm run lint`.